src/duplo_resource/s3.py

```diff
@@ -0,0 +1,73 @@
+from duplocloud.client import DuploClient
+from duplocloud.resource import DuploTenantResourceV2
+from duplocloud.errors import DuploError
+from duplocloud.commander import Command, Resource
+import duplocloud.args as args
+
+
+@Resource("s3")
+class DuploS3(DuploTenantResourceV2):
+  
+  def __init__(self, duplo: DuploClient):
+    super().__init__(duplo)
+
+  @Command()
+  def list(self):
+    """Retrieve a list of all infrastructures in the Duplo system."""
+    tenant_id = self.tenant["TenantId"]
+    tenant_name = self.tenant["AccountName"]
+    response = self.duplo.get(f"v3/subscriptions/{tenant_id}/aws/s3Bucket")
+    if (data := response.json()):
+      return data
+    else:
+      raise DuploError(f"No s3 bucket found in tenant '{tenant_name}'", 404)
+    
+  
+  @Command()
+  def create(self, 
+             body: args.BODY,
+             wait: args.WAIT = False):
+    """Create a new s3."""
+    def wait_check():
+      name = self.name_from_body(body)
+      self.find(name)
+    tenant_id = self.tenant["TenantId"]
+    self.duplo.post(f"v3/subscriptions/{tenant_id}/aws/s3Bucket", body)
```

Comment:
Most of the Resource base classes have an `endpoint` method to get the proper endpoint for the particular class. The v3 apis especially need this as they have a very strict pattern that can be followed. You probably won't even need to do this once you properly extend the `DuploTenantResourceV3` because the methods already handle requests. 

This is the endpoint method for v3 resources: 
https://github.com/duplocloud/duploctl/blob/cb2c0d7bfcca5918e054939e08d4bc90f64ce3f0/src/duplocloud/resource.py#L195

Notice how the parent `update` command uses the endpoint method: 
https://github.com/duplocloud/duploctl/blob/cb2c0d7bfcca5918e054939e08d4bc90f64ce3f0/src/duplocloud/resource.py#L189